packages/components/src/Combobox/components/ComboboxContent/ComboboxContentHeader/ComboboxContentHeader.tsx

```diff
@@ -0,0 +1,50 @@
+import React from "react";
+import { Typography } from "@jobber/components/Typography";
+import { Button } from "@jobber/components/Button";
+import { ComboboxHeaderProps } from "@jobber/components/Combobox/Combobox.types";
+import styles from "./ComboboxContentHeader.css";
+
+export function ComboboxContentHeader(props: ComboboxHeaderProps): JSX.Element {
+  const hasSelected = props.selectedCount > 0;
+  const actionLabel = hasSelected ? "Clear" : "Select all";
+  const label = getLabel(hasSelected, props.selectedCount, props.subjectNoun);
+  const handleSelectAll = hasSelected ? props.onClearAll : props.onSelectAll;
+
+  const searchTermMatches = props.options.some(option =>
+    option.label.toLowerCase().includes(props.searchValue.toLowerCase()),
+  );
+
+  const showClearButton = hasSelected || searchTermMatches;
```

Comment:
I don't mind the name but what I do think we can do differently here is leverage work we've already done!

in `ComboboxContent` we already have a `filteredOptions` that is doing pretty much this exact operation to figure out if we have any matches

what we could do instead is 

```
 <ComboboxContentHeader
          subjectNoun={props.subjectNoun}
          selectedCount={selectedOptions.length}
          onClearAll={() => {
            optionsSelectionHandler([]);
          }}
          onSelectAll={() => {
            optionsSelectionHandler(filteredOptions);
          }}
          hasResults={filteredOptions.length > 0}
          searchValue={searchValue}
          options={props.options}
        />
```

and then add the prop `readonly hasResults: boolean;` to our types of course

this way we can cut down on iterations, DRY up the code and yeah just take advantage of work we've already done within the component!

and then I suppose to better match the `hasSelected` a name like `hasResults` or `hasSearchResults` might fit just a tiny bit better

Reply:
another little benefit from this is that we no longer have to pass in `searchValue` as a prop to this

although I think the odds of reusing this might be low, it's nice to be able to de-couple those concepts. the less we need a component to know about the others, or our specific implementation - the better!

if we thought this was going to be generalized, there's even more we could do - but for now, I think letting it be just a tad bit more naive is a nice improvement